python/lsst/cp/pipe/cpDarkTask.py

```diff
@@ -106,26 +109,52 @@ def run(self, inputExp):
                                         self.config.psfSize,
                                         self.config.psfFwhm/(2*math.sqrt(2*math.log(2))))
         inputExp.setPsf(psf)
-        scaleExp = inputExp.clone()
-        mi = scaleExp.getMaskedImage()
 
-        # DM-23680:
-        # Darktime scaling necessary for repair.run() to ID CRs correctly.
-        scale = inputExp.getInfo().getVisitInfo().getDarkTime()
-        if np.isfinite(scale) and scale != 0.0:
-            mi /= scale
+        gains = self._get_gains(inputExp)
+        with gainContext(inputExp, inputExp.getVariance(), apply=True, gains=gains):
+            self.log.info("Median image and variance: %f %f",
+                          np.median(inputExp.image.array), np.median(inputExp.variance.array))
+            try:
+                self.repair.run(inputExp, keepCRs=False)
+            except LengthError:
+                self.log.warning("CR rejection failed!")
+            self.log.info("Number of CR pixels: %d",
+                          np.count_nonzero(np.bitwise_and(inputExp.mask.array,
+                                                          inputExp.mask.getPlaneBitMask('CR'))))
 
-        self.repair.run(scaleExp, keepCRs=False)
         if self.config.crGrow > 0:
             crMask = inputExp.getMaskedImage().getMask().getPlaneBitMask("CR")
             spans = afwGeom.SpanSet.fromMask(inputExp.mask, crMask)
             spans = spans.dilated(self.config.crGrow)
+            spans = spans.clippedTo(inputExp.getBBox())
```

Comment:
This is needed because `SpanSet`s are generic, and have no image information attached.  Without this, CR pixels at the edge of the image can be dilated outside the image extent, and that causes the `setMask` operation to raise.